src/server/coprocessor/executor/table_scan.rs

```diff
@@ -0,0 +1,231 @@
+
+    struct TableScanExecutorMeta {
+        data: Data,
+        store: TestStore,
+        table_scan: TableScan,
+        ranges: Vec<KeyRange>,
+        cols: Vec<ColumnInfo>,
+    }
+
+    impl Default for TableScanExecutorMeta {
+        fn default() -> TableScanExecutorMeta {
+            let test_data = prepare_table_data(KEY_NUMBER, TABLE_ID);
+            let test_store = TestStore::new(&test_data.data, test_data.pk.clone());
+            let mut table_scan = TableScan::new();
+            // prepare cols
+            let cols = test_data.get_prev_2_cols();
+            let col_req = RepeatedField::from_vec(cols.clone());
+            table_scan.set_columns(col_req);
+            // prepare range
+            let range = get_range(TABLE_ID, i64::MIN, i64::MAX);
+            let key_ranges = vec![range];
+            TableScanExecutorMeta {
+                data: test_data,
+                store: test_store,
+                table_scan: table_scan,
+                ranges: key_ranges,
+                cols: cols,
+            }
+        }
+    }
+
+    #[test]
+    fn test_point_get() {
+        let mut statistics = Statistics::default();
+        let mut meta = TableScanExecutorMeta::default();
+        // prepare range
+        let mut range = KeyRange::new();
+        range.set_start(meta.data.pk.clone());
+        let end = prefix_next(&meta.data.pk.clone());
+        range.set_end(end);
+        assert!(is_point(&range));
+        meta.ranges = vec![range];
+        let mut table_scanner = TableScanExec::new(meta.table_scan,
+                                                   meta.ranges,
+                                                   meta.store.store(),
+                                                   &mut statistics);
+
+        let row = table_scanner.next().unwrap().unwrap();
+        assert_eq!(row.handle, meta.data.pk_handle);
+        assert_eq!(row.data.len(), meta.cols.len());
+
+        let encode_data = &meta.data.encode_data[0];
+        for col in &meta.cols {
+            let cid = col.get_column_id();
+            let v = row.data.get(cid).unwrap();
+            assert_eq!(encode_data[&cid], v.to_vec());
+        }
+        assert!(table_scanner.next().unwrap().is_none());
+    }
+
+    #[test]
+    fn test_multiple_ranges() {
+        let mut statistics = Statistics::default();
+        let mut meta = TableScanExecutorMeta::default();
+        // prepare range
+        let r1 = get_range(TABLE_ID, i64::MIN, 0);
+        let r2 = get_range(TABLE_ID, 0, (KEY_NUMBER / 2) as i64);
+        let r3 = get_range(TABLE_ID, (KEY_NUMBER / 2) as i64, i64::MAX);
+        meta.ranges = vec![r1, r2, r3];
+
+        let mut table_scanner = TableScanExec::new(meta.table_scan,
+                                                   meta.ranges,
+                                                   meta.store.store(),
+                                                   &mut statistics);
+
+        for handle in 0..KEY_NUMBER {
```

Comment:
It's easy to be confused with `handle` and `row.handle`. Any better name for `handle`?